src/core/array_slice.pm

```diff
@@ -221,37 +289,37 @@ multi sub postcircumfix:<[ ]>(\SELF, Callable:D $block, :$SINK!, *%other) is rw
     SLICE_MORE_LIST( SELF, POSITIONS(SELF,$block), :$SINK, |%other );
 }
 multi sub postcircumfix:<[ ]>(\SELF,Callable:D $block,:$delete!,*%other) is rw {
-    my $pos := $block(|(SELF.elems xx $block.count));
+    my $pos := $block(|(SELF.elems xx ($block.count == Inf ?? 1 !! $block.count)));
     nqp::istype($pos,Int)
       ?? SLICE_ONE_LIST(  SELF,  $pos, :$delete, |%other )
       !! SLICE_MORE_LIST( SELF, @$pos, :$delete, |%other );
 }
 multi sub postcircumfix:<[ ]>(\SELF,Callable:D $block,:$exists!,*%other) is rw {
-    my $pos := $block(|(SELF.elems xx $block.count));
+    my $pos := $block(|(SELF.elems xx ($block.count == Inf ?? 1 !! $block.count)));
     nqp::istype($pos,Int)
       ?? SLICE_ONE_LIST(  SELF,  $pos, :$exists, |%other )
       !! SLICE_MORE_LIST( SELF, @$pos, :$exists, |%other );
 }
 multi sub postcircumfix:<[ ]>(\SELF, Callable:D $block, :$kv!, *%other) is rw {
-    my $pos := $block(|(SELF.elems xx $block.count));
+    my $pos := $block(|(SELF.elems xx ($block.count == Inf ?? 1 !! $block.count)));
```

Comment:
I think it would better to write that as `my $count = $block.count; my $pos := $count == Inf || $block == 1 ?? $block(SELF.elems) ?? $block(|(SELF.elems xx $count)`

(and likewise in the other multis).

postcircumfixes are a rather hot path, and we should avoid the infix:<xx> call for a single element, as well as the double `.count` calls for the common path (.count == 1).